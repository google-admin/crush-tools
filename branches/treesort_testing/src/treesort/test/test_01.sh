test_number=01
description="two column sort"

subtest_desc=("DD,BB"
							"3,1 using index"
							"^DD,BB"
							"^DD,^BB"
							"4,^2 -p")
subtest_opts=(
'-K DD,BB'
'-k 3,1 -p'
'-K ^DD,BB'
'-K ^DD,^BB'
'-k 4,^2 -p'
)

for subtest in `seq 0 4`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
