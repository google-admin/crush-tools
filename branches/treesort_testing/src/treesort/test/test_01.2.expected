DDþBBþAAþCC
32.4092þ3XbaqX1keSþ843þ2009-10-02
32.4092þ5TcvIxjquUþ577þ2009-10-15
32.4092þOCS81RVo5jþ222þ2009-10-28
32.4092þtHGNJyQcu1þ278þ2009-10-01
29.5253þvOXGmYLegsþ468þ2009-10-04
28.4036þA-cSNvcqnzþ468þ2009-10-01
28.4036þZfcHogunHrþ843þ2009-10-01
28.4036þarzhFJlrMBþ577þ2009-10-14
26.6454þ6ff-bhImN7þ918þ2009-10-10
22.8364þOcNUM48IS6þ740þ2009-10-15
20.7444þWDEGX5smiWþ279þ2009-10-15
20.7444þafUGgY6q4jþ279þ2009-10-14
20.7444þz3cqeuch4gþ918þ2009-10-28
17.0268þ0dGw4UOK0Pþ279þ2009-10-26
17.0268þ7YaqxPKpCRþ740þ2009-10-01
17.0268þW2OIToxqAMþ843þ2009-10-04
17.0268þxVciPqOBiQþ965þ2009-10-14
14.9256þENItEY_hIBþ991þ2009-10-23
14.3872þ4j4n56w__lþ577þ2009-10-25
13.5003þ49Uj8LVH5jþ222þ2009-10-02
